os/include/tinyara/binfmt/elf.h

```diff
@@ -163,6 +163,7 @@ struct elf_loadinfo_s {
 	uint16_t buflen;			/* size of iobuffer[] */
 	int filfd;					/* Descriptor for the file being loaded */
 	uint16_t offset;             /* elf offset when binary header is included */
+	uint8_t compression_type;		/* Binary Compression type */
```

Comment:
Include this variable only if CONFIG_COMPRESSED_BINARY is enabled?

Reply:
elf loadinfo struct needs to have compression type.

Reply:
According to info provided to libelf about if binary is compressed/uncompressed and compression_type, it will handle binary read/load.